cmd/server/app/webhook_update.go

```diff
@@ -96,47 +89,54 @@ func runCmdWebhookUpdate(cmd *cobra.Command, _ []string) error {
 		return fmt.Errorf("unable to parse webhook url: %w", err)
 	}
 
-	fallbackTokenClient := ghprovider.NewFallbackTokenClient(cfg.Provider)
+	whSecret, err := getWebhookSecret(cfg)
+	if err != nil {
+		return err
+	}
+
+	providerManager, err := wireUpProviderManager(cfg, store)
+	if err != nil {
+		return fmt.Errorf("failed to instantiate provider manager: %w", err)
+	}
 
 	for _, provider := range allProviders {
-		zerolog.Ctx(ctx).Info().Str("name", provider.Name).Str("uuid", provider.ID.String()).Msg("provider")
-		pb, err := providers.GetProviderBuilder(ctx, provider, store, cryptoEng, &cfg.Provider, fallbackTokenClient)
-		if err != nil {
-			return fmt.Errorf("unable to get provider builder: %w", err)
+		if providerName != "" && providerName != provider.Name {
+			continue
 		}
 
-		if !pb.Implements(db.ProviderType(providerName)) {
+		if !provider.CanImplement(db.ProviderTypeGithub) {
+			// currently we can only operate on GitHub
+			// revisit this once we add more providers with webhooks
 			zerolog.Ctx(ctx).Info().
 				Str("name", provider.Name).
 				Str("uuid", provider.ID.String()).
 				Msg("provider does not implement the requested provider interface")
 			continue
 		}
 
-		var updateErr error
-
-		if db.ProviderType(providerName) == db.ProviderTypeGithub {
-			ghCli, err := pb.GetGitHub()
-			if err != nil {
-				zerolog.Ctx(ctx).Err(err).Msg("cannot get github client")
-				continue
-			}
-
-			whSecret, err := cfg.WebhookConfig.GetWebhookSecret()
```

Comment:
As far as I can tell, this is constant across all iterations of the loop, so I moved it outside of the loop